tests/integration/devfile/cmd_dev_test.go

```diff
@@ -1398,4 +1415,52 @@ var _ = Describe("odo dev command tests", func() {
 			}
 		})
 	})
+
+	When("a component with multiple endpoints is run", func() {
+		stateFile := ".odo/devstate.json"
+		var devSession helper.DevSession
+		BeforeEach(func() {
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project-with-multiple-endpoints"), commonVar.Context)
+			helper.Cmd("odo", "project", "set", commonVar.Project).ShouldPass()
+			helper.Cmd("odo", "init", "--name", cmpName, "--devfile-path", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile-with-multiple-endpoints.yaml")).ShouldPass()
+			Expect(helper.VerifyFileExists(".odo/devstate.json")).To(BeFalse())
+			var err error
+			devSession, _, _, _, err = helper.StartDevMode()
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			// We stop the process so the process does not remain after the end of the tests
+			devSession.Kill()
+			devSession.WaitEnd()
+		})
+
+		It("should create a state file containing forwarded ports", func() {
+			Expect(helper.VerifyFileExists(stateFile)).To(BeTrue())
+			contentJSON, err := ioutil.ReadFile(stateFile)
+			Expect(err).ToNot(HaveOccurred())
+			helper.JsonPathContentIs(string(contentJSON), "forwardedPorts.0.containerName", "runtime")
+			helper.JsonPathContentIs(string(contentJSON), "forwardedPorts.1.containerName", "runtime")
+			helper.JsonPathContentIs(string(contentJSON), "forwardedPorts.0.localAddress", "127.0.0.1")
+			helper.JsonPathContentIs(string(contentJSON), "forwardedPorts.1.localAddress", "127.0.0.1")
+			helper.JsonPathContentIs(string(contentJSON), "forwardedPorts.0.containerPort", "3000")
+			helper.JsonPathContentIs(string(contentJSON), "forwardedPorts.1.containerPort", "4567")
+			helper.JsonPathContentMatch(string(contentJSON), "forwardedPorts.0.localPort", "[0-9]+")
```

Comment:
`[0-9]+` could potentially match `0` or even negative values, I think. I think we should really test for a port strictly greater than 0.
Maybe using a regexp like `^[1-9]+[0-9]*$`.

We could even go further by also testing that the number written for the port is less than 65,535.